src/connectedk8s/azext_connectedk8s/_troubleshootutils.py

```diff
@@ -0,0 +1,802 @@
+        cli_outputs.append("An expection has occured while performing msi expiry check on the cluster. Exception: {}".format(str(e)) + "\n")
+
+    return "Incomplete"
+
+
+def describe_stuck_agent_log(filepath_with_timestamp, corev1_api_instance, agent_pod_name, storage_space_available):
+
+    try:
+        if storage_space_available:
+            describe_stuck_agent_path = os.path.join(filepath_with_timestamp, 'Describe_Stuck_Agents')
+            try:
+                os.mkdir(describe_stuck_agent_path)
+            except FileExistsError:
+                pass
+            api_response = corev1_api_instance.read_namespaced_pod(name=agent_pod_name, namespace='azure-arc')
+            stuck_agent_pod_path = os.path.join(describe_stuck_agent_path, agent_pod_name + '.txt')
+
+            with open(stuck_agent_pod_path, 'w+') as stuck_agent_log:
+                stuck_agent_log.write(str(api_response))
+
+    except OSError as e:
+        if "[Errno 28]" in str(e):
+            storage_space_available = False
+            telemetry.set_exception(exception=e, fault_type=consts.No_Storage_Space_Available_Fault_Type, summary="No space left on device")
+            shutil.rmtree(filepath_with_timestamp, ignore_errors=False, onerror=None)
+        else:
+            logger.warning("An expection has occured while storing stuck agent logs in the user local machine. Exception: {}".format(str(e)) + "\n")
+            telemetry.set_exception(exception=e, fault_type=consts.Describe_Stuck_Agents_Fault_Type, summary="Error occured while storing the stuck agents description")
+            cli_outputs.append("An expection has occured while storing stuck agent logs in the user local machine. Exception: {}".format(str(e)) + "\n")
+
+    except Exception as e:
+            logger.warning("An expection has occured while storing stuck agent logs in the user local machine. Exception: {}".format(str(e)) + "\n")
+            telemetry.set_exception(exception=e, fault_type=consts.Describe_Stuck_Agents_Fault_Type, summary="Error occured while storing the stuck agents description")
+            cli_outputs.append("An expection has occured while storing stuck agent logs in the user local machine. Exception: {}".format(str(e)) + "\n")
+
+    return storage_space_available
+
+
+def cli_output_logger(filepath_with_timestamp, storage_space_available):
+
+    global cli_outputs
+
+    try:
+
+        if storage_space_available:
+
+            cli_output_logger_path = os.path.join(filepath_with_timestamp, "Diagnoser_Results.txt")
+
+            if len(cli_outputs) > 0:
+                with open(cli_output_logger_path, 'w+') as cli_output_writer:
+                    for output in cli_outputs:
+                        cli_output_writer.write(output + "\n")
+            else:
+                with open(cli_output_logger_path, 'w+') as cli_output_writer:
+                        cli_output_writer.write("Diagnoser could not find any issues with the cluster.\n")
```

Comment:
nit: Change this sentence to "diagnoser did not fine any issues with the cluster"